resources-domain/resources-service/scripts/khp/importLocationsCsv.ts

```diff
@@ -0,0 +1,154 @@
+ON CONFLICT DO NOTHING;
+INSERT INTO resources."ResourceReferenceStringAttributeValues" ("accountSid", "list", "id", "value", "language", "info") VALUES ($<accountSid>, 'provinces', $<idFr>, $<value>, 'fr', $<infoFr>)
+ON CONFLICT DO NOTHING;
+UPDATE resources."ResourceReferenceStringAttributes" SET "referenceId" = $<id> WHERE "accountSid" = $<accountSid> AND "list" = 'provinces' AND "referenceId" = $<oldId>;
+UPDATE resources."ResourceReferenceStringAttributes" SET "referenceId" = $<idFr> WHERE "accountSid" = $<accountSid> AND "list" = 'provinces' AND "referenceId" = $<oldIdFr>;
+DELETE FROM resources."ResourceReferenceStringAttributeValues" WHERE "accountSid" = $<accountSid> AND "list" = 'provinces' AND "id" IN ($<oldId>, $<oldIdFr>);
+`,
+        {
+          accountSid,
+          id: `CA-${geographicCode}-en`,
+          idFr: `CA-${geographicCode}-fr`,
+          value: `CA/${code}`,
+          info: { name, geographicCode },
+          infoFr: { name: CANADIAN_PROVINCE_CODE_FR_MAP[code] },
+          oldId: `CA-${code}-en`,
+          oldIdFr: `CA-${code}-fr`,
+        },
+      ),
+    );
+    provincesJson.push({
+      label: name,
+      value: `CA/${code}`,
+    });
+  });
+  sqlFile.write('\n\n--- CITIES ---\n\n');
+  for await (const line of csvLines) {
+    const [geographicCode, cityEn, cityFr, csdType, , province] = line as string[];
+    const [provinceCode] = CANADIAN_PROVINCE_NAME_CODE_MAP[
+      province as keyof typeof CANADIAN_PROVINCE_NAME_CODE_MAP
+    ];
+    const sqlStatement = pgp.as.format(
+      `
+INSERT INTO resources."ResourceReferenceStringAttributeValues" ("accountSid", "list", "id", "value", "language", "info") VALUES ($<accountSid>, 'cities', $<id>, $<value>, 'en', $<info>)
+ON CONFLICT DO NOTHING;;
+INSERT INTO resources."ResourceReferenceStringAttributeValues" ("accountSid", "list", "id", "value", "language", "info") VALUES ($<accountSid>, 'cities', $<idFr>, $<value>, 'fr', $<infoFr>)
+ON CONFLICT DO NOTHING;
+UPDATE resources."ResourceReferenceStringAttributes" SET "referenceId" = $<id> WHERE "accountSid" = $<accountSid> AND "list" = 'cities' AND "referenceId" = $<oldId>;
+UPDATE resources."ResourceReferenceStringAttributes" SET "referenceId" = $<idFr> WHERE "accountSid" = $<accountSid> AND "list" = 'cities' AND "referenceId" = $<oldIdFr>;
+DELETE FROM resources."ResourceReferenceStringAttributeValues" WHERE "accountSid" = $<accountSid> AND "list" = 'provinces' AND "id" IN ($<oldId>, $<oldIdFr>);
+`,
+      {
+        accountSid: process.argv[2],
+        id: `CA-${geographicCode}-en`,
+        idFr: `CA-${geographicCode}-fr`,
+        value: `CA/${provinceCode}/${cityEn}`,
+        info: { name: cityEn, geographicCode, csdType },
+        infoFr: { name: cityFr, geographicCode, csdType },
+        oldId: `CA-${provinceCode}-${cityEn}-en`,
+        oldIdFr: `CA-${provinceCode}-${cityEn}-fr`,
```

Comment:
These old `ResourceReferenceStringAttributeValues` that we're deleting are just:
- out of date values; or
- data in wrong format (id in old format); or
- both of the above; or
- neither 😅 (Maybe I'm just a bit lost because I don't have the ticket context).

Suppose KHP gives us an updated locations CSV file (same structure, different data values). Do we need to rewrite these lines (129, 130)?

Reply:
They are just old values imported when we were using a different scheme for our internal IDs because KHP didn't provide the codes

Those lines can probably go in the future, I kept them around just in case but they should be redundant now if we get data in the new format from now on